ophyd/sim.py

```diff
@@ -375,8 +375,17 @@ def position(self):
         return self.readback.get()
 
 
-class SynAxis(SynAxisNoHints):
-    readback = Component(ReadbackSignal, value=None, kind=Kind.hinted)
+class SynAxisEmptyHints(SynAxis):
```

Comment:
It isn't enough to just lower all the Components to `omitted` in the test?

`motor.readback.kind = 'omitted'`

Seems strange to add a class to the library just for this test?

Reply:
Yes, it sounds reasonable. I just wanted to follow the existing approach when the `SynAxisNoHints` class already existed (however was wrongly defined.) I tried `motor.readback.kind = 'omitted'`, however it fails on Bluesky's BestEffortCallback, as in this case `motor.hints` returns `{'field': []}`, not `{}`. We will probably have to change how it's handled by Bluesky. I'll post the code and traceback into the linked issue there.

Reply:
Isn't that the more realistic behavior? It appears to me the only way you'll get a completely empty dictionary is a malformed subclass with `hints` defined.

https://github.com/bluesky/ophyd/blob/be880e1bcc87c05668bf300845e32416537e647b/ophyd/device.py#L1261